### PR TITLE
Add checkpoint size metrics

### PR DESCRIPTION
**Pull request by zhangchiqing**
Close https://github.com/onflow/flow-go/issues/5392

### CODE REVIEW

File: module/metrics/execution.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ type ExecutionCollector struct {
 	lastFinalizedExecutedBlockHeightGauge   prometheus.Gauge
 	stateStorageDiskTotal                   prometheus.Gauge
 	storageStateCommitment                  prometheus.Gauge
+	checkpointSize                          prometheus.Gauge
```
Comment (janezpodhostnik) [inline module/metrics/execution.go:25]: Does this need an initialisation?

File: module/metrics/execution.go [hunk 2]
```diff
@@ -650,6 +651,7 @@ func NewExecutionCollector(tracer module.Tracer) *ExecutionCollector {
 			Help:      "the execution state size on disk in bytes",
 		}),
 
+		// TODO: remove
```
Comment (janezpodhostnik) [inline module/metrics/execution.go:654]: Is this todo for another PR? If so, maybe add more context.

File: module/metrics.go [hunk 3]
```diff
@@ -729,6 +729,8 @@ type LedgerMetrics interface {
 }
 
 type WALMetrics interface {
+	// ExecutionCheckpointSize reports the size of a checkpoint in bytes
+	ExecutionCheckpointSize(bytes int64)
```
Comment (janezpodhostnik) [inline module/metrics.go:733]:
```suggestion
	ExecutionCheckpointSize(bytes uint64)
```

**Review by janezpodhostnik — APPROVED** [review]

File: ledger/complete/compactor.go [hunk 4]
```diff
@@ -337,6 +341,15 @@ func createCheckpoint(checkpointer *realWAL.Checkpointer, logger zerolog.Logger,
...
 
+	size, err := realWAL.ReadCheckpointFileSize(checkpointer.Dir(), fileName)
+	if err != nil {
+		return fmt.Errorf("error reading checkpoint file size (%d): %w", checkpointNum, err)
+	}
+
+	metrics.ExecutionCheckpointSize(size)
+
+	// metrics.ExecutionSize
+
```
Comment (peterargue) [inline ledger/complete/compactor.go:352]:
```suggestion
```

File: ledger/complete/wal/checkpoint_v6_reader.go [hunk 5]
```diff
@@ -105,6 +105,34 @@ func OpenAndReadCheckpointV6(dir string, fileName string, logger zerolog.Logger)
...
+		}
+
+		totalSize += fileInfo.Size()
+	}
+
+	return totalSize, nil
+}
+
+func allFilePaths(dir string, fileName string) []string {
+	paths := make([]string, 1+subtrieCount+1)
```
Comment (peterargue) [inline ledger/complete/wal/checkpoint_v6_reader.go:125]:
```suggestion
	paths := make([]string, 0, 1+subtrieCount+1)
```

File: ledger/complete/wal/checkpointer.go [hunk 6]
```diff
@@ -252,7 +253,14 @@ func (c *Checkpointer) Checkpoint(to int) (err error) {
...
 	}
 
-	c.wal.log.Info().Msgf("created checkpoint %d with %d tries", to, len(tries))
+	checkpointFileSize, err := ReadCheckpointFileSize(c.wal.dir, fileName)
+	if err != nil {
+		return fmt.Errorf("could not read checkpoint file size: %w", err)
+	}
+
+	c.wal.log.Info().
+		Str("checkpoint file size", units.BytesSize(float64(checkpointFileSize))).
```
Comment (peterargue) [inline ledger/complete/wal/checkpointer.go:262]: make it easier to work with labels as metrics in grafana
```suggestion
		Str("checkpoint_file_size", units.BytesSize(float64(checkpointFileSize))).
```

File: module/metrics/execution.go (hunk 1 above)
Comment (peterargue) [reply module/metrics/execution.go:25]: yes, I think it will panic if it's not initialized

**Review by peterargue — APPROVED** [review]

**Pull request merged**